elide-core/src/main/java/com/yahoo/elide/core/filter/dialect/RSQLFilterDialect.java

```diff
@@ -0,0 +1,315 @@
+        path = Paths.get(path).normalize().toString().replace(File.separatorChar, '/');
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+
+        String[] pathComponents = path.split("/");
+        String lastPathComponent = "";
+        if (pathComponents.length > 0) {
+            lastPathComponent = pathComponents[pathComponents.length - 1];
+        }
+
+        Class entityType = dictionary.getEntityClass(lastPathComponent);
+        if (entityType == null) {
+            throw new ParseException("No such collection: " + lastPathComponent);
+        }
+
+        try {
+            Node ast = parser.parse(queryParamValue);
+            RSQL2ExpressionFilterVisitor visitor = new RSQL2ExpressionFilterVisitor(true);
+            FilterExpression filterExpression = ast.accept(visitor, entityType);
+            return filterExpression;
+        } catch (RSQLParserException e) {
+            throw new ParseException(e.getMessage());
+        }
+    }
+
+    @Override
+    public Map<String, FilterExpression> parseTypedExpression(String path, MultivaluedMap<String, String>
+            queryParams) throws ParseException {
+
+        Map<String, FilterExpression> expressionByType = new HashMap<>();
+
+        for (MultivaluedMap.Entry<String, List<String>> entry : queryParams.entrySet()) {
+
+            String paramName = entry.getKey();
+            List<String> paramValues = entry.getValue();
+
+            Matcher matcher = typedFilterPattern.matcher(paramName);
+            if (matcher.find()) {
+                String typeName = matcher.group(1);
+                if (paramValues.size() != 1) {
+                    throw new ParseException("Exactly one RSQL expression must be defined for type : " + typeName);
+                }
+
+                String expressionText = paramValues.get(0);
+
+                Class entityType = dictionary.getEntityClass(typeName);
+
+                if (entityType == null) {
+                    throw new ParseException("Invalid query parameter: " + paramName);
+                }
+
+                try {
+                    Node ast = parser.parse(expressionText);
+                    RSQL2ExpressionFilterVisitor visitor = new RSQL2ExpressionFilterVisitor(false);
+                    FilterExpression filterExpression = ast.accept(visitor, entityType);
+                    expressionByType.put(typeName, filterExpression);
+                } catch (RSQLParserException e) {
+                    throw new ParseException(e.getMessage());
+                }
+            } else {
+                throw new ParseException("Invalid query parameter: " + paramName);
+            }
+        }
+        return expressionByType;
+    }
+
+    /**
+     * Visitor which converts RSQL abstract syntax tree into an Elide filter expression.
+     */
+    public class RSQL2ExpressionFilterVisitor implements RSQLVisitor<FilterExpression, Class> {
```

Comment:
Can we move this out to its own file? The constructor would also have to take a dictionary (and operatorMap if we don't make it static), but it may be easier to digest.